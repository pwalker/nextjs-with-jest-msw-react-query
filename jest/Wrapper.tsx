import { ReactNode } from "react";
import { QueryClient, QueryClientProvider, setLogger } from "react-query";

setLogger({
  log: console.log,
  warn: console.warn,
  // ✅ no more errors on the console
  error: () => {},
});

export const Wrapper: React.FC<{ children: ReactNode }> = ({ children }) => {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        // ✅ turns retries off
        retry: false,
      },
    },
  });
  return (
    <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
  );
};
