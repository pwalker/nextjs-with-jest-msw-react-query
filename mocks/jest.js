import { rest } from "msw";
import { setupServer } from "msw/node";

export const server = setupServer(
  rest.get("/greeting", (req, res, ctx) => {
    console.log("greetings!");
    return res(ctx.json({ greeting: "hello there" }));
  })
);
