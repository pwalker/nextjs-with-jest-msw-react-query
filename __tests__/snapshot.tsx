import { render } from "@testing-library/react";
import Home from "@/pages/index";
import { Wrapper } from "../jest/Wrapper";

it("renders homepage unchanged", () => {
  const { container } = render(
    <Wrapper>
      <Home />
    </Wrapper>
  );
  expect(container).toMatchSnapshot();
});
