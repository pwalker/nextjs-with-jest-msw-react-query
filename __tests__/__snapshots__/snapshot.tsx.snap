// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <div
    class="container"
  >
    <h1>
      Welcome to the test app!
    </h1>
    <div>
      loading
    </div>
  </div>
</div>
`;
